py/qstr.c

```diff
@@ -234,11 +264,13 @@ STATIC qstr qstr_add(mp_uint_t hash, mp_uint_t len, const char *q_ptr) {
 
 qstr qstr_find_strn(const char *str, size_t str_len) {
     // work out hash of str
+    #if MICROPY_QSTR_BYTES_IN_HASH
     size_t str_hash = qstr_compute_hash((const byte *)str, str_len);
+    #endif
 
     // search pools for the data
     for (const qstr_pool_t *pool = MP_STATE_VM(last_pool); pool != NULL; pool = pool->prev) {
-        size_t low = 0;
+        size_t low = pool->prev ? 0 : 1; // skip MP_QSTRnull at the start of the first pool.
```

Comment:
I'm not sure it's worth adding this logic.  It's not needed (you'll never get to this point with `str_len == 0` so the check agains the length in the search will never match `MP_QSTRnull` or `MP_QSTR_`) and adds an extra if/jump in a somewhat critical loop.  Skipping just 1 entry in a binary search also won't make it noticeably faster.  It also increases code size.

Reply:
4212799fd8bded69e5cf47faf2ebd675906ac151

Reply:
Yes I think I solved this in two different ways and didn't catch that in the rebase.